BLE_SupportedFeatures/source/main.cpp

```diff
@@ -0,0 +1,164 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <events/mbed_events.h>
+#include "ble/BLE.h"
+#include "ble/gap/Gap.h"
+#include "pretty_printer.h"
+
+using namespace std::literals::chrono_literals;
+
+const static char DEVICE_NAME[] = "SupportedBLEFeatures";
+
+static events::EventQueue event_queue(/* event count */ 16 * EVENTS_EVENT_SIZE);
+
+class SupportedFeatures : ble::Gap::EventHandler {
+public:
+    SupportedFeatures(BLE &ble, events::EventQueue &event_queue) :
+        _ble(ble),
+        _event_queue(event_queue),
+        _service_uuid(GattService::UUID_DEVICE_INFORMATION_SERVICE),
+        _adv_data_builder(_adv_buffer)
+    {
+    }
+
+    void start()
+    {
+        _ble.init(this, &SupportedFeatures::on_init_complete);
+
+        _event_queue.dispatch_forever();
+    }
+
+private:
+    /** Callback triggered when the ble initialization process has finished */
+    void on_init_complete(BLE::InitializationCompleteCallbackContext *params)
+    {
+        if (params->error != BLE_ERROR_NONE) {
+            printf("Ble initialization failed.");
+            return;
+        }
+
+        print_mac_address();
+
+        /* display basic info and optional BLE features supported by our device */
+
+        printf("BLE Version %s\n", _ble.getVersion());
+        printf("Conn handle %d\n", ble::connection_handle_t(&_ble));
```

Comment:
This case doesn't make sense. What was your intention here?

Reply:
These can be removed, i just tried to fetch as much data about the BLE instance as possible.